viewing_party/party.py

```diff
@@ -1,23 +1,126 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dictionary = {}
+    if title and genre and rating:
+        movie_dictionary["title"] = title
+        movie_dictionary["genre"] = genre
+        movie_dictionary["rating"] = rating
+    else:
+        return None
+    
+    return movie_dictionary
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    copy_user_data = user_data
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            copy_user_data["watched"].append(movie)
+            copy_user_data["watchlist"].remove(movie)
+        user_data = copy_user_data
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    sum = 0
+    length = len(user_data["watched"])
+
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            sum += movie["rating"]
+        avg_rating = sum / length
+    
+    return avg_rating
+def get_most_watched_genre(user_data):
+    dict_count_genre = {}
+    if user_data["watched"]:
+       for movie in user_data["watched"]:
+           if not movie["genre"] in dict_count_genre:
+              dict_count_genre[movie["genre"]] = 1
+           else:
+              dict_count_genre[movie["genre"]] += 1

```

Comment:
Nice! Another way you could build up a frequency dictionary for genre would be to use the .get() with a default value, which lets us get away from an if/else loop to either set the value to 1 or increment the value by 1.

```python
for movie in user_data['watched']: 
    current_genre = movie["genre"]
    dict_count_genre[current_genre] = dict_count_genre.get(current_genre, 0) + 1
```